.evergreen/generate-evergreen-yml.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+const fs = require('fs');
+
+// PHP-like syntax, except it's JS.
+// foo: <% for (let i = 0; i < 10; i++) { %>
+//   - index: <% out(i) %>
+// <% } %>
+// is turned into a YAML list with index: 0, index: 1 etc.
+
+const input = fs.readFileSync(process.argv[2], 'utf8');
+let asJs = `(function() { let result = "";
+  const out = (val) => result += (typeof val === 'object' && val !== null) ? JSON.stringify(val) : val;`;
+let i = 0;
+while (i < input.length) {
+  const opening = input.indexOf('<%', i);
+  if (opening === -1) {
+    asJs += `out(${JSON.stringify(input.slice(i))});\n`;
+    break;
+  }
+  asJs += `out(${JSON.stringify(input.slice(i, opening).replace(/\n\s*$/, ''))});\n`;
+  const closing = input.indexOf('%>', opening + 2);
+  if (closing === -1) break;
+  asJs += input.slice(opening + 2, closing) + ';\n';
+
+  i = closing + 2;
+}
+asJs += '; return result; })()'
+process.stdout.write(eval(asJs));
```

Comment:
Happy to take feedback about this approach :)

Reply:
FWIW I'd assume something like handlebars is a more commonly used template tool for this kind of stuff, but yeah, you won't be able to have variables in the same file (at least I think it's not supported?) and it's another dependency. This solves the problem, just 28 lines, and zero dependencies so I'm totally fine with this custom template method 😊

Reply:
Yeah, I thought about using handlebars, but … felt a bit more flexible this way? :)

Reply:
Totes! Love the PHP vibes also 😄